sklearn/gaussian_process/kernels.py

```diff
@@ -1686,8 +1686,9 @@ def __call__(self, X, Y=None, eval_gradient=False):
                 D = squareform(dists**2)[:, :, np.newaxis]
 
             if self.nu == 0.5:
-                K_gradient = K[..., np.newaxis] * D \
-                    / np.sqrt(D.sum(2))[:, :, np.newaxis]
+                denominator = np.sqrt(D.sum(2))[:, :, np.newaxis]
+                denominator[denominator == 0.0] = 1e-8
+                K_gradient = K[..., np.newaxis] * D / denominator
```

Comment:
I find it a bit cleaner to only divide where it's non zero
```py
denominator = np.sqrt(D.sum(2))[:, :, np.newaxis]
K_gradient = K[..., np.newaxis] * np.divide(D, denominator, where=denominator != 0)
```

Reply:
Won't it change the normalisation?

Reply:
D = 0 where denominator = 0, so K_gradient = 0 where denominator = 0, regardless of the normalization

Reply:
Yeah, that is much cleaner way.
I applied your suggestion, thanks.